src/printers/flow.js

```diff
@@ -19,11 +19,67 @@ export default {
     ? Tokens.string('true')
     : Tokens.string('false'),
 
+  ExistentialTypeParam: () => [
+    Tokens.string('*'),
+  ],
+
+  FunctionTypeAnnotation: ({node, print, path}) => [
+    print(node.typeParameters),
+    Tokens.string('('),
+    (node.params.length || node.rest) && [
+      Tokens.scopeOpen('object'),
+      Tokens.scopeEmptyOrBreak(),
+      map(node.params, (param, i, arr) => [
+        i > 0 && Tokens.scopeSpaceOrBreak(),
+        print(param),
+        arr.length - 1 !== i && !node.rest
+          ? Tokens.comma()
+          : Tokens.scopeEmptyOrComma(),
+      ]),
+      node.rest && [
+        node.params.length && [
+          Tokens.comma(),
+          Tokens.scopeSpaceOrBreak(),
+        ],
+        Tokens.string('...'),
+        print(node.rest),
+        Tokens.scopeEmptyOrComma(),
+      ],
+      Tokens.scopeEmptyOrBreak(),
+      Tokens.scopeClose(),
+    ],
+    Tokens.string(')'),
+    (
+      path[path.length - 2].type === 'ObjectTypeProperty' ||
```

Comment:
Yeah I think it's okay to reach up and find parent from child node sometimes. But generally it's better to do it in another way if it's possible and reasonably clean. So this is fine. I guess I'm not sure if I want to expose parent so easily, I feel like it will start to get used in more places. But maybe that's okay too /shrug